src/backend/distributed/executor/multi_utility.c

```diff
@@ -257,20 +253,17 @@ multi_ProcessUtility(Node *parsetree,
 	 */
 	if (EnableDDLPropagation)
 	{
-		bool isTopLevel = (context == PROCESS_UTILITY_TOPLEVEL);
-
 		if (IsA(parsetree, IndexStmt))
 		{
-			parsetree = ProcessIndexStmt((IndexStmt *) parsetree, queryString,
-										 isTopLevel);
+			ddlJob = PlanIndexStmt((IndexStmt *) parsetree, queryString);
```

Comment:
Was starting out consistent and can carve out modifications as needed during the review. I agree that there are probably only certain commands that need planning first but was keeping changes to mostly shuffling things around in as least invasive a way possible.